platforms/jvm/testing-jvm/src/main/java/org/gradle/api/tasks/testing/Test.java

```diff
@@ -203,6 +202,10 @@ public Object call() {
         javaLauncher.finalizeValueOnRead();
         getDryRun().convention(false);
         testFramework = objectFactory.property(TestFramework.class).convention(objectFactory.newInstance(JUnitTestFramework.class, this.getFilter(), this.getTemporaryDirFactory(), this.getDryRun()));
+
+        getTestDefinitionDirs().from(getProject().getLayout().getProjectDirectory().file(TEST_DEFINITIONS_DIR));
```

Comment:
This needs to be defined outside the task.  We shouldn't be calling `getProject` in a task

Reply:
It looks like other tasks inject `ProjectLayout` directly.  I'll switch to an internal property and avoid accessing the `project` at construction time.